aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -500,19 +500,29 @@ void TensorIterator::select_all_keeping_dim(int start_dim, IntArrayRef indices)
   }
 }
 
-TensorIterator TensorIterator::binary_op(Tensor& out, const Tensor& a, const Tensor& b) {
+TensorIterator TensorIterator::binary_op(Tensor& out, const Tensor& a,
+    const Tensor& b, bool check_internal_overlap) {
   auto iter = TensorIterator();
-  iter.add_output(out);
+  if (check_internal_overlap) {
+    iter.add_checked_output(out);
```

Comment:
nit: This name can be confusing because technically the function does the checking. How about something like `check_and_add_output`?

Reply:
renamed